card-tricks/skoved/card_tricks.go

```diff
@@ -0,0 +1,53 @@
+package cards
+
+// isValidIndex determines if index is a valid position in slice
+func isValidIndex(slice []int, index int) bool {
+	return index >= 0 && index < len(slice)
+}
+
+// GetItem retrieves an item from a slice at given position. The second return value indicates whether
+// the given index exists in the slice or not.
+func GetItem(slice []int, index int) (int, bool) {
+	if isValidIndex(slice, index) {
+		return slice[index], true
+	}
+
+	return 0, false
+}
+
+// SetItem writes an item to a slice at given position overwriting an existing value.
+// If the index is out of range the value needs to be appended.
+func SetItem(slice []int, index, value int) []int {
+	if isValidIndex(slice, index) {
+		slice[index] = value
+	} else {
+		slice = append(slice, value)
+	}
+	return slice
+}
+
+// PrefilledSlice creates a slice of given length and prefills it with the given value.
+func PrefilledSlice(value int, length int) []int {
+	// This step of the excersize didn't give insight into handling negative length, but the test
+	// case for negative length looks for a slice with no elements
+	if length < 0 {
```

Comment:
You don't need this check with a normal for loop. The check is included in it. Something like this:
```go
func PrefilledSlice(value, length int) (slice []int) {
	for i := 0; i < length; i++ {
		slice = append(slice, value)
	}

	return
}
```

Reply:
Using the named return value makes this a lot cleaner